seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,57 +1,133 @@
+    if not workspace_has_perm(request.user, CAN_EDIT, namespace, name, can_share=True):
+        error = "Missing required permissions for loading data from workspace {}/{}".format(namespace, name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    add_service_account(request.user, namespace, name)
+
+    # Parse families/individuals in the uploaded pedigree file
+    try:
+        json_records = load_uploaded_file(request_json['uploadedFileId'])
+    except Exception as ee:
+        error = "Uploaded pedigree file is missing or other exception: {}".format(str(ee))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    pedigree_records, errors, ped_warnings = parse_pedigree_table(json_records, 'ped_file', user=request.user, project=project)
+    if errors:
+        error = "Parse pedigree data failed. {}".format(errors)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Create a new Project in seqr
+    project_args = {
+        'name': project_name,
+        'genome_version': request_json['genomeVersion'],
+        'description': request_json.get('description', ''),
+        'workspace_namespace': namespace,
+        'workspace_name': name,
+    }
+
+    project = create_model_from_json(Project, project_args, user = request.user)
+
+    # update families and individuals according to the uploaded individual records
+    updated_families, updated_individuals = _add_or_update_individuals_and_families(
+        project, individual_records = pedigree_records, user = request.user
+    )
+
+    # Send an email to all seqr data managers
+    info = ['{} families and {} individuals have been added to the new project'.format(len(updated_families),
+                                                                                       len(updated_individuals))]
+    individual_ids = [individual['individualId'] for individual in pedigree_records]
+    try:
+        send_load_data_email(request.user, project.guid, namespace, name, individual_ids)
+    except Exception as ee:
+        message = 'Exception while sending email to user {}. {}'.format(request.user, (ee))
+        logger.error(message)
+        info.append(message)
```

Comment:
this should never be returned to the user. The user does not know or care about what we are doing behind the scenes